Assignment6.Rmd

```diff
@@ -63,21 +65,45 @@ post(c.tree2, file = "tree2.ps", title = "MOOC") #This creates a pdf image of th
 
 ```{r}
 M2$predict1 <- predict(c.tree1, M2, type = "class")
-
 M2$predict2 <- predict(c.tree2, M2, type = "class")
 
-table(M2$certified, M2$predict1)
+M2$error1 <- ifelse(M2$certified == M2$predict1, 0, 1)
+M2$error2 <- ifelse(M2$certified == M2$predict2, 0, 1)
 
-table(M2$certified, M2$predict2)
+error_rate1 <- sum(M2$error1)/nrow(M2)
+error_rate2 <- sum(M2$error2)/nrow(M2)
 
+# The pruned tree has a lower error rate.
 ```
 
 ##Part III
 
 Choose a data file from the (University of Michigan Open Data Set)[https://github.com/bkoester/PLA/tree/master/data]. Choose an outcome variable that you would like to predict. Build two models that predict that outcome from the other variables. The first model should use raw variables, the second should feature select or feature extract variables from the data. Which model is better according to the cross validation metrics?
 
 ```{r}
+S1 <- read.csv("student.course.csv", header = TRUE)
+
+# Predict GPAO from the other variables
+c.tree3 <- rpart(GPAO ~ . - GPAO, data = S1)
+printcp(c.tree3)
+post(c.tree3, file = "tree3.ps")
+
+
+# Predict GPAO from TERM, SUBJECT and DIV
+c.tree4 <- rpart(GPAO ~ TERM + SUBJECT + DIV, data = S1)
+printcp(c.tree4)
+post(c.tree4, file = "tree4.ps")
+
+# The first model resulted in 7 splits, with a cross validation error of .6763. The second model resulted in 2 splits, with a cross validation error of .9876.
+
+# According to the graph of model 2 (tree.4), only SUBJECT variable could provide information when predicting GPAO. This could be checked by predicting GPAO only from TERM or from DIV. As shown below, both of them had only one split.
+c.tree5 <- rpart(GPAO ~ TERM, data = S1)
+printcp(c.tree5)
+
+c.tree6 <- rpart(GPAO ~ DIV, data = S1)
+printcp(c.tree6)
 
+# Based on the cross validation metrics, the first model was better because it had a smaller cross validation error. This might be because the more variables are used for prediction, the smaller the error is. 
```

Comment:
I like the way you tried many times with different sets of predictors/variables and compare between them. Instead of checking TEAM and DIV in univariate models, why not just look at the model with SUBJECT? Besides, it would be great if you could give some interpretation of your final model.